File: Solutions In Java/p204.java
/* 
 * Solution to Project Euler problem 204
 * Copyright (c) Sankha Saha (CODR). All rights reserved.
 * 
 * https://github.com/codr07/Project-Euler-Solusions
 */


public final class p204 implements EulerSolution {
	
	public static void main(String[] args) {
		System.out.println(new p204().run());
	}
	
	
	public String run() {
		return Integer.toString(count(0, 1));
	}
	
	
	private static long LIMIT = Library.pow(10, 9);
	
	private int[] primes = Library.listPrimes(100);
	
	
	private int count(int primeIndex, long product) {
		if (primeIndex == primes.length)
			return product <= LIMIT ? 1 : 0;
		else {
			int count = 0;
			while (product <= LIMIT) {
				count += count(primeIndex + 1, product);
				product *= primes[primeIndex];
			}
			return count;
		}
	}
	
}

File: Solutions in Python/p204.py
# 
# Solution to Project Euler problem 204
# Copyright (c) Sankha Saha (CODR). All rights reserved.
# 
#
# https://github.com/codr07/Project-Euler-Solusions
# 

import eulerlib


def compute():
	LIMIT = 10**9
	primes = eulerlib.list_primes(100)
	
	def count(primeindex, product):
		if primeindex == len(primes):
			return 1 if product <= LIMIT else 0
		else:
			result = 0
			while product <= LIMIT:
				result += count(primeindex + 1, product)
				product *= primes[primeindex]
			return result
	
	return str(count(0, 1))


if __name__ == "__main__":
	print(compute())
